Update Cloud Costs page and add Cloud release note (#19076)
cockroachdb/docs: src/current/cockroachcloud/costs.md

For CockroachDB Standard and Advanced, CockroachDB Cloud bills directly to customers based on their usage with a zero-dollar rate during the preview period.

The usage data is based on the volume of data moved (per GiB) and the cloud provider’s identified data transfer type, which CockroachDB categorizes as **Same-region data transfer**, **Cross-Region data transfer**, or **Public Internet data transfer**. Each category is reflected as a line item on your invoice.

> Cloud providers may vary in how they calculate each category. We recommend consulting the provider’s documentation for details (i.e. [AWS](https://aws.amazon.com/ec2/pricing/on-demand/), [GCP](https://cloud.google.com/vpc/network-pricing), or [Azure](https://azure.microsoft.com/en-us/pricing/details/bandwidth/)).

Reviewer: ```suggestion
Cloud providers vary in how they calculate each category. Consult the provider’s documentation for details. For example:
- [AWS EC2 on-demand pricing](https://aws.amazon.com/ec2/pricing/on-demand/)
- [GCP network pricing](https://cloud.google.com/vpc/network-pricing)
- [Azure bandwidth pricing](https://azure.microsoft.com/en-us/pricing/details/bandwidth/)
```

#### Same-region

> There are no charges for same-region data transfer between CockroachDB cluster nodes. Data egressing the cluster to endpoints in the same region for backups or changefeeds, or using private connectivity, is charged the cloud provider's list price for same-region data transfer.

Reviewer: ```suggestion
There is no charge for same-region data transfer between cluster nodes in CockroachDB Cloud. Backup or changefeed data that egresses from the cluster to an endpoint in the same region, or that is sent over private networks, is charged the cloud provider's list price for same-region data transfer.
```
I took a stab at trying to improve this. Ignore if this is not what you mean.

#### Cross-region

> When data moves between CockroachDB nodes that are in different regions, or from a CockroachDB node to another endpoint in a different region, it is metered as cross-region data transfer. 

Reviewer: ```suggestion
When data moves between cluster nodes in different regions, or when it egresses from a CockroachDB node to an endpoint in a different region, it is charged as cross-region data transfer. This includes:
```

> Cross-region data transfer includes:

Reviewer: ```suggestion
```

> - Data transfer required to support queries that involve lookups on nodes in another region.  

Reviewer: This has a lot of indirection and could be rewritten a bit to avoid "required to support" and "involve". Maybe "Queries that read or write data on a node in a different region" maybe?

> - CockroachDB replication across nodes that are in different regions.  

Reviewer: ```suggestion
- Replication from one node to another node in a different region.
```

> - Data egress from the CockroachDB Cloud cluster via supported [private connectivity](https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster#establish-private-connectivity) services to a private endpoint in another region.  

Reviewer: ```suggestion
- Data egressing from the cluster over [private networks](https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster#establish-private-connectivity) to an endpoint in a different region.  
```

> - [Managed backup](https://www.cockroachlabs.com/docs/cockroachcloud/backup-and-restore-overview#managed-backups) and [Self-managed backup](https://www.cockroachlabs.com/docs/cockroachcloud/take-and-restore-self-managed-backups) data transfer to another region.  

Reviewer: ```suggestion
- [Managed backups](https://www.cockroachlabs.com/docs/cockroachcloud/backup-and-restore-overview#managed-backups) and [Self-managed backup](https://www.cockroachlabs.com/docs/cockroachcloud/take-and-restore-self-managed-backups) that transfer data from one region to another.  
```

> - Change Data Capture (changefeed) data transfer to another region.

Reviewer: ```suggestion
- Change Data Capture (changefeed) data transfer from one region to another.
```

> Customers are charged the cloud service provider’s (CSP) list price for metered cross-region data transfer.

Reviewer: ```suggestion
Cross-region data transfer is charged at the cloud provider’s list price for metered cross-region data transfer.
```
Is it worth linking to the official cloud provider rates here instead of at the end of the section?

> For CockroachDB Standard and Advanced, CockroachDB Cloud bills directly to customers based on their usage with a zero-dollar rate during the preview period.

Reviewer: Standard and Advanced should have separate tabs. I am having a hard time parsing this sentence.

> The usage data is based on the volume of data moved (per GiB) and the cloud provider’s identified data transfer type, which CockroachDB categorizes as **Same-region data transfer**, **Cross-Region data transfer**, or **Public Internet data transfer**. Each category is reflected as a line item on your invoice.

Reviewer: "usage data"? Do youy mean "Usage is based on..."? Maybe "the type of data transfer, as identified by the Cloud provider. Data transfer is classified as ...."

Cloud providers may vary in how they calculate each category. We recommend consulting the provider’s documentation for details (i.e. [AWS](https://aws.amazon.com/ec2/pricing/on-demand/), [GCP](https://cloud.google.com/vpc/network-pricing), or [Azure](https://azure.microsoft.com/en-us/pricing/details/bandwidth/)).

#### Same-region

> There are no charges for same-region data transfer between CockroachDB cluster nodes. Data egressing the cluster to endpoints in the same region for backups or changefeeds, or using private connectivity, is charged the cloud provider's list price for same-region data transfer.

Reviewer: Similar feedback as similar text earlier in the page for the sections below.

- Data transfer required to support queries that involve lookups on nodes in another region.  
- CockroachDB replication across nodes that are in different regions.  
- Data egress from the CockroachDB Cloud cluster via supported [private connectivity](https://www.cockroachlabs.com/docs/cockroachcloud/connect-to-your-cluster#establish-private-connectivity) services to a private endpoint in another region.  
- [Managed backup](https://www.cockroachlabs.com/docs/cockroachcloud/backup-and-restore-overview#managed-backups) and [Self-managed backup](https://www.cockroachlabs.com/docs/cockroachcloud/take-and-restore-self-managed-backups) data transfer to another region.  
- Change Data Capture (changefeed) data transfer to another region.

Customers are charged the cloud service provider’s (CSP) list price for metered cross-region data transfer.

Typically, data transfer costs across regions are higher than data transfer costs across AZs (data centers) in a single region.

#### Public internet

This is the usage for any data leaving CockroachDB such as SQL data being sent to clients, or logs and metrics exported to third-party tools.

</section>

### Change Data Capture (Changefeeds)

> All CockroachDB {{ site.data.products.cloud }} clusters can use [Enterprise Changefeeds]({% link {{ site.current_cloud_version}}/how-does-an-enterprise-changefeed-work.md %}).

Reviewer: ```suggestion
[Enterprise Changefeeds]({% link {{ site.current_cloud_version}}/how-does-an-enterprise-changefeed-work.md %}) are supported on all clusters in CockroachDB Cloud.
```

## Usage-based billing metrics in Preview
<a id="usage-based-costs"></a>

> Metering for usage-based billing of data transfer, managed backup storage, and changefeeds is in [Preview](https://www.cockroachlabs.com/docs/stable/cockroachdb-feature-availability) for all CockroachDB Cloud organizations.

Reviewer: Sometimes we talk about "costs", "charges", or "metering". It might be good to pick one way so that it doesn't look like we are trying to differentiate between them when I suspect we mean the same thing by all of them.

- Usage metrics for data transfer, managed backup storage, and changefeeds are now visible for CockroachDB Standard and Advanced clusters in the CockroachDB Cloud Console. You can view your usage across these metrics on the [Billing page](https://cockroachlabs.cloud/billing/overview) and on invoices.  

> - There will be no usage-based charges associated with these metrics during the Preview period, which is in effect through November 30, 2024. During this time, line items with a charge of $0 will be shown for each metric on your monthly invoice.  

Reviewer: ```suggestion
- There will be no usage-based charges associated with these metrics during the Preview period, which is in effect through November 30, 2024. During the Preview, line items will be shown for each metric with a charge of $0 on your monthly invoice.  
```

- Future pricing for these usage-based costs is available on the [Pricing](https://www.cockroachlabs.com/pricing) page.  

> - On December 1, 2024, once the Preview has ended, pricing for these metrics goes into effect immediately for new customers and for existing pay-as-you-go customers (e.g. paying monthly by credit card). Customers with annual or multi-year contracts will continue to preview these line items without incurring charges for them (i.e. expending credits) through the end of their current contract term.

Reviewer: How about Marketplace?
